inflate.c

```diff
@@ -1050,12 +1050,25 @@ int32_t Z_EXPORT PREFIX(inflate)(PREFIX3(stream) *strm, int32_t flush) {
                 if (INFLATE_NEED_CHECKSUM(strm) && state->wrap & 4) {
                     if (out) {
                         inf_chksum(strm, put - out, out);
+                    } else if ((!state->flags && strm->adler == ADLER32_INITIAL_VALUE)
+#ifdef GUNZIP
+                    || (state->flags && state->crc_fold.value == CRC32_INITIAL_VALUE)) {
+#else
+                        ) {
+#endif
+                        /* In the event we reach this block and the checksums are at the initial values, run a checksum
+                         * over the whole stream, as one of two things are true:
+                         *  1.) We never ran any of the bytes through the checksum (testing has found this happens)
+                         *  2.) We have a checksum that does wrap perfectly, in which case, we should end up still
+                         *      correct, anyway */
+                        inf_chksum(strm, put, strm->total_out);
                     }
 #ifdef GUNZIP
                     if (state->flags)
                         strm->adler = state->check = functable.crc32_fold_final(&state->crc_fold);
 #endif
                 }
+
```

Comment:
Adding this empty line seems unnecessary... Mixing white-space and code changes is frowned upon...